mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMavericksViewModel.kt

```diff
@@ -164,6 +169,84 @@ abstract class BaseMavericksViewModel<S : MvRxState>(
         stateStore.get(action)
     }
 
+    /**
+     * Run a coroutine and wrap its progression with [Async] property reduced to the global state.
+     *
+     * @param dispatcher The coroutine dispatcher that the coroutine will run on. Defaults to [Dispatchers.Main.immediate].
+     * @param retainValue A state property that, when set, will be called to retrieve an optional existing data value that will be retained across
+     *                    subsequent Loading and Fail states. This is useful if you want to display the previously succcessful data when
+     *                    refreshing.
+     * @param reducer A reducer that is applied to the current state and should return the new state. Because the state is the receiver
+     *                and it likely a data class, an implementation may look like: `{ copy(response = it) }`.
+     */
+    fun <T : Any?> Deferred<T>.execute(
+        dispatcher: CoroutineDispatcher = Dispatchers.Main.immediate,
```

Comment:
@gpeal @BenSchwab what if we pull the dispatcher out of the viewmodelscope instead and use that as the default? something like
```
dispatcher: CoroutineDispatcher = viewModelScope.coroutineContext[ContinuationInterceptor]
```

The ContinuationInterceptor is used as the context key for the CoroutineDispatcher class which should let us pull out the dispatcher. This would let us default to whichever dispatcher the scope uses

Reply:
since coroutines 1.3.x CoroutineDispatcher is coroutine context key too, so it can be pulled directly:
```
viewModelScope.coroutineContext[CoroutineDispatcher]
```